gtsam/base/GenericValue.h

```diff
@@ -89,21 +89,17 @@ class GenericValue: public Value {
 
     /**
      * Create a duplicate object returned as a pointer to the generic Value interface.
-     * For the sake of performance, this function use singleton pool allocator instead of the normal heap allocator.
-     * The result must be deleted with Value::deallocate_, not with the 'delete' operator.
      */
     virtual Value* clone_() const {
-      void *place = boost::singleton_pool<PoolTag, sizeof(GenericValue)>::malloc();
-      GenericValue* ptr = new (place) GenericValue(*this); // calls copy constructor to fill in
+      GenericValue* ptr = new GenericValue(*this); // calls copy constructor to fill in
```

Comment:
Perhaps there is a way to expand `EIGEN_MAKE_ALIGNED_OPERATOR_NEW` to cope with this additional overload of `new`, but it would probably be a mess and not worth the alleged performance gain of this `singleton_pool`... 
Unless you profiled that the singleton_pool is **really** worth in some application (?).

Reply:
This is a placement new so the object does not have any control over alignment... also the pool only have `malloc` so no guarantees on alignment whatsoever. I can do a manual alignment here but I don't think this is worth it. And there is no proof of the alleged "performance gain" either :)

Reply:
> there is no proof of the alleged "performance gain" either :)

Not true. One of Frank's former students profiled the pool allocator when it was first introduced, and it made a significant difference. Please profile before you remove.

Reply:
I meant "performance gain" due to lack of continuous malloc/free vs. using the pool. It all depends on how many times you need to call this, I guess...

> This is a placement new so the object does not have any control over alignment...

Oh! Then the alternative would be checking if singleton_pool<> allows for aligned memory blocks, not changing the Eigen macros, you're right.

Reply:
> Oh! Then the alternative would be checking if singleton_pool<> allows for aligned memory blocks, not changing the Eigen macros, you're right.

I have investigated this before. It does not support aligned allocation in itself, so we need to manually allocate `sizeof(object)+alignment` of memory and shift the pointer a bit to the aligned position.

> Not true. One of Frank's former students profiled the pool allocator when it was first introduced, and it made a significant difference. Please profile before you remove.

You are right that using an arena allocator makes performance gains in some workloads. However, I think the proper way of doing this would be using stuff like `jemalloc` which have been field proven and have proper `aligned_alloc()` support. `boost::singleton_pool` is just another arena allocator, but it is not widely used (google only have 3000 results).